src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -176,24 +184,69 @@ function ToolbarButton({
 type TextAreaProps = {
   classes?: string;
   containerRef?: Ref<HTMLTextAreaElement>;
+  atMentionsEnabled: boolean;
 };
 
 function TextArea({
   classes,
   containerRef,
+  atMentionsEnabled,
   ...restProps
 }: TextAreaProps & JSX.TextareaHTMLAttributes<HTMLTextAreaElement>) {
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const textareaRef = useSyncedRef(containerRef);
+
+  useEffect(() => {
+    if (!atMentionsEnabled) {
+      return () => {};
+    }
+
+    const textarea = textareaRef.current!;
+    const listenerCollection = new ListenerCollection();
+
+    // We listen for `keyup` to make sure the text in the textarea reflects the
+    // just-pressed key when we evaluate it
+    listenerCollection.add(textarea, 'keyup', e => {
+      // `Esc` key is used to close the popover. Do nothing and let users close
+      // it that way, even if the caret is in a mention
+      if (e.key === 'Escape') {
+        return;
+      }
+      setPopoverOpen(
+        termBeforePosition(textarea.value, textarea.selectionStart).startsWith(
+          '@',
+        ),
+      );
```

Comment:
I'm splitting it in the "next" branch already, because I need to save the actual term in a piece of state, so I'm going to keep it like this here to reduce conflicts.